src/core/src/pass/sdpa_to_paged_attention.cpp

```diff
@@ -149,19 +149,25 @@ bool ov::pass::SDPAToPagedAttention::run_on_model(const std::shared_ptr<ov::Mode
         }
     }
 
-    for (auto& param_name : {"beam_idx", "attention_mask"}) {
+    for (const auto& param_name : {"beam_idx", "attention_mask"}) {
         if (auto param = get_parameter(model, param_name)) {
```

Comment:
As far as I remember, 
the `attention_mask` input usually has rank == 2, e.g. [dyn, dyn] in Qwen

attention_mask (rank == 2) -> shape_of ->

Initially `input_ids` has rank == 2, but after applying this transformation it will have  rank == 1 
so after this change, we can get

input_ids (rank == 1) -> shape_of

This can lead to implicit/explicit issues if there is a shape-dependent operation after this `shape_of`, e.g. gather

input_ids (rank == 1) -> shape_of -> gather

We use unsqueeze for input_ids (line 84) to be align with the original layout, should we use `unsqueezed_input_ids` here?

Reply:
Discussed privately.
I connect ShapeOf to the unsqueezed version of the updated input_ids. This way we stay consistent with the transformation approach.